pkgs/development/libraries/gstreamer/gstreamermm/default.nix

```diff
@@ -1,26 +1,63 @@
-{ lib, stdenv, fetchurl, pkg-config, file, glibmm, gst_all_1, gnome }:
+{ lib
+, stdenv
+, fetchFromGitLab
+, autoconf
+, automake
+, libtool
+, perl
+, mm-common
+, pkg-config
+, file
+, glibmm
+, gst_all_1
+, nix-update-script
+}:
+
 stdenv.mkDerivation rec {
   pname = "gstreamermm";
-  version = "1.10.0";
+  version = "unstable-2022-05-03";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "0q4dx9sncqbwgpzma0zvj6zssc279yl80pn8irb95qypyyggwn5y";
+  src = fetchFromGitLab {
+    # https://gitlab.gnome.org/GNOME/gstreamermm
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gstreamermm";
+    # https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/5
```

Comment:
I think I'd much prefer if this was applied via a patches = [ ... ];

Currently this gives the impression (at least to me) that this is just backporting some already accepted upstream patch. But in fact, this seems to still be waiting for review and merge into the upstream repo.

Reply:
I can merge it upstream. But it is a good idea to give people chance to review when it is not my project.

Reply:
> nitpickery ... also, patches would bloat nixpkgs

I did not mean this as a nitpick, I strongly value making derivation sources auditable and easy to understand. Replacing this with a fetchpatch of https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/5.diff or something similar makes it significantly clearer that this is unmerged WIP changes.

> yeah. why should we wait? upstream is an old project, maintainers are not responsive

I'm not at all suggesting to wait, nor am I opposed to having these kind of patches for nixpkgs derivations. But if we carry "custom" patches I think we should properly document it as such.

Reply:
> if we carry "custom" patches I think we should properly document it as such.

that's why i added the comment https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/5 to `fetchFromGitLab`

> it is a good idea to give people chance to review when it is not my project.

the patch is trivial, build is passing. testing the runtime is on my todo list ...

risk: we merge this today as `unstable-2022-05-03`
and tomorrow upstream merges your patch as `1.10.1`
then we would need 1 more commit to sync with upstream

but i guess that risk is low,
and upstream will take the "usual" weeks and months to merge this
let's wait 2 more days for upstream?